From c0a672aec2ca0fb9bc321d3ecc2460b89a0f82f9 Mon Sep 17 00:00:00 2001
From: Anson Huang <Anson.Huang@nxp.com>
Date: Tue, 12 Dec 2017 21:07:43 +0800
Subject: [PATCH 3085/5242] MLK-17188-2 ARM64: dts: imx8qxp: assign usdhc
 clock parent

commit  a5a7a9c30b375038a1fcbbc659c696d205577882 from
https://source.codeaurora.org/external/imx/linux-imx.git

Assign i.MX8QXP uSDHC clocks parent to from PLL1.

This is a workaround for i.MX8QXP usdhc, PLL0 of CONN SS is not
stable sometimes, root cause is still under investigation in
design team. Now change to source from PLL1. Due to PLL1 is
1000MHz, so EMMC HS400ES mode can only work at 166MHz, compare
to the former 198MHz, the performance has small drop, read
performance drop about 10%, write performance drop about 6%.
SD do not has this side effect.

When PLL0 unstable issue is fixed, will change back to use PLL0.

Signed-off-by: Anson Huang <Anson.Huang@nxp.com>
Reviewed-by: Bai Ping <ping.bai@nxp.com>
Tested-by: Haibo Chen <haibo.chen@nxp.com>
Signed-off-by: Meng Li <Meng.Li@windriver.com>
---
 arch/arm64/boot/dts/freescale/fsl-imx8qxp.dtsi |   15 +++++++++------
 1 file changed, 9 insertions(+), 6 deletions(-)

diff --git a/arch/arm64/boot/dts/freescale/fsl-imx8qxp.dtsi b/arch/arm64/boot/dts/freescale/fsl-imx8qxp.dtsi
index cdb4bd2..f00d9e0 100644
--- a/arch/arm64/boot/dts/freescale/fsl-imx8qxp.dtsi
+++ b/arch/arm64/boot/dts/freescale/fsl-imx8qxp.dtsi
@@ -2311,8 +2311,9 @@
 			<&clk IMX8QXP_SDHC0_CLK>,
 			<&clk IMX8QXP_CLK_DUMMY>;
 		clock-names = "ipg", "per", "ahb";
-		assigned-clocks = <&clk IMX8QXP_SDHC0_DIV>;
-		assigned-clock-rates = <400000000>;
+		assigned-clocks = <&clk IMX8QXP_SDHC0_SEL>, <&clk IMX8QXP_SDHC0_DIV>;
+		assigned-clock-parents = <&clk IMX8QXP_CONN_PLL1_CLK>;
+		assigned-clock-rates = <0>, <400000000>;
 		power-domains = <&pd_conn_sdch0>;
 		fsl,tuning-start-tap = <20>;
 		fsl,tuning-step= <2>;
@@ -2328,8 +2329,9 @@
 			<&clk IMX8QXP_SDHC1_CLK>,
 			<&clk IMX8QXP_CLK_DUMMY>;
 		clock-names = "ipg", "per", "ahb";
-		assigned-clocks = <&clk IMX8QXP_SDHC1_DIV>;
-		assigned-clock-rates = <200000000>;
+		assigned-clocks = <&clk IMX8QXP_SDHC1_SEL>, <&clk IMX8QXP_SDHC1_DIV>;
+		assigned-clock-parents = <&clk IMX8QXP_CONN_PLL1_CLK>;
+		assigned-clock-rates = <0>, <200000000>;
 		power-domains = <&pd_conn_sdch1>;
 		fsl,tuning-start-tap = <20>;
 		fsl,tuning-step= <2>;
@@ -2345,8 +2347,9 @@
 			<&clk IMX8QXP_SDHC2_CLK>,
 			<&clk IMX8QXP_CLK_DUMMY>;
 		clock-names = "ipg", "per", "ahb";
-		assigned-clocks = <&clk IMX8QXP_SDHC2_DIV>;
-		assigned-clock-rates = <200000000>;
+		assigned-clocks = <&clk IMX8QXP_SDHC2_SEL>, <&clk IMX8QXP_SDHC2_DIV>;
+		assigned-clock-parents = <&clk IMX8QXP_CONN_PLL1_CLK>;
+		assigned-clock-rates = <0>, <200000000>;
 		power-domains = <&pd_conn_sdch2>;
 		status = "disabled";
 	};
-- 
1.7.9.5

